Composite/Composite/Transforms.fs

```diff
@@ -35,10 +35,44 @@ module Transforms =
                     while cur |> Option.isSome do
                         yield cur |> (function | Some x -> x | None -> failwith "Unexpected situation in enumeration.")
                         cur <- getNext ()
-                }
+            }
 
         Array.init numParts (fun _ -> getSeq ())
 
+    let toPaged pageSize inSeq =
+        let rec getPageAndRest numRemains pageAndRest = 
+            match numRemains > 0, pageAndRest with
+            | true, (p, r) -> match r |> Seq.tryHead with
+                                | Some h -> getPageAndRest (numRemains-1) (Array.append p [|h|], r |> Seq.tail)
+                                | None -> pageAndRest
+            | false, pr -> pr
+
+        let rec getPages inSeq2 =
+            seq {
+                    match getPageAndRest pageSize ([||], inSeq2) with
+                    | [||], _ -> yield! []
+                    | p, r -> yield p
+                              yield! getPages r
+            }
+        getPages inSeq
+
+    let toBatched batchSize getElemSize inSeq =
+        let rec getBatchAndRest numRemains pageAndRest = 
+            match numRemains > 0, pageAndRest with
+            | true, (p, r) -> match r |> Seq.tryHead with
+                                | Some h -> getBatchAndRest (numRemains-(getElemSize h)) (Array.append p [|h|], r |> Seq.tail)

```

Comment:
A little style point. The parameter 'p' may be passed to Array.append by the pipe-forward operator while appending a new element to a batch.
The method toPaged may be changed in the same way.
Also, the value pageAndRest may be renamed to batchAndRest as it does in #15.